### PR TITLE
Update e3sm to compilersv2

### PR DESCRIPTION
**Pull request by jgfouca**
Remove v1 support too.

Allow nesting within SHELL items in config_compilers.xml

Remove support for tag syntax since nesting works fine now.

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: N

Update gh-pages html (Y/N)?: N

Code review: @jedwards4b

### CODE REVIEW

File: config/e3sm/machines/config_compilers.xml [hunk 1]
```diff
@@ -78,24 +77,158 @@ for mct, etc.
...
+  </LDFLAGS>
   <SUPPORTS_CXX>FALSE</SUPPORTS_CXX>
-  <ADD_CPPDEFS MODEL="pop"> -D_USE_FLOW_CONTROL </ADD_CPPDEFS>
 </compiler>
 
-
 <!-- hacking of mach/compiler generated 'Macros.make' for coupling with pflotran -->
 <!-- ideally it should go with CLM configuration -->
-<compiler>
-  <ADD_FFLAGS  MODEL="clm"    CLM_PFLOTRAN_COUPLED="TRUE"> -I$(CLM_PFLOTRAN_SOURCE_DIR) </ADD_FFLAGS>
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply config/e3sm/machines/config_compilers.xml]: This code should not be in config_compilers.xml at all.  The appropriate place for it is in the
scripts/Tools/Makefile in a manor similar to that of USE_PETSC or USE_TRILINOS

[the remark this answers is not in the document]
Comment (jgfouca) [reply config/e3sm/machines/config_compilers.xml]: Thanks, I figured this was an abuse of the XML.

**Comment by jedwards4b** [comment]
Travis-ci is failing in compilerblock.py

**Comment by jgfouca** [comment]
@jedwards4b , fixed. Let me make the Makefile changes before you merge.

**Comment by jgfouca** [comment]
@jedwards4b , I done a significant update to this PR: removing support for <tag> syntax in config_compilers. It was only necessary because the previous implementation did not support nesting the ${} syntax. That works now. CESM was only using it in one spot which I have updated.

**Comment by jgfouca** [comment]
Ready for review but hold off on merge.

**Comment by jgfouca** [comment]
@jedwards4b , because ENV{} works just a well.

**Pull request merged**